flask_project/campaign_manager/views.py

```diff
@@ -1039,48 +911,127 @@ def create_campaign():
+                creator_id=new_creator.id,
+                start_date=data['start_date'],
+                end_date=data['end_date'],
+                create_on=datetime.now(),
+                uuid=data['new_uuid'],
+                version=2,
+                map_type=data['map_type']
+            )
+            campaign_copy.create()
+            campaign_copy.save_feature_types(data)
+            campaign_copy.save_geometry(data)
+            campaign_copy.save_insight_functions(data)
+            campaign_copy.save_managers(data)
+            create_thumbnail_image(campaign_copy)
+            return redirect(
+                    url_for('campaign_manager.get_campaign',
+                            uuid=campaign_copy.uuid)
+                    )
+    context['url'] = '/edit/copy/%s' % uuid
+    context['mode'] = 'copy'
+    context['campaign'] = campaign_obj
+    try:
+        context['types'] = json.dumps(
+            get_types()).replace('True', 'true').replace('False', 'false')
+    except ValueError:
+        pass
+    return render_template(
+        'create_campaign.html', form=form, **context)
+
+
 @campaign_manager.route('/edit/<uuid>', methods=['GET', 'POST'])
 def edit_campaign(uuid):
+    import uuid as _uuid
     from flask import url_for, redirect
     from campaign_manager.forms.campaign import CampaignForm
     """Get campaign details.
     """
-    context = {}
     try:
         campaign = Campaign()
         campaign_obj = campaign.get_by_uuid(uuid)
-        context['campaign'] = campaign_obj
-        managers = campaign_obj.users
-        managers = [x.osm_user_id for x in managers]
-        campaign_types = {}
-
-        # Preprocess campaign data
-        for _ in campaign_obj.feature_types:
-            campaign_types[_.name] = {}
-            campaign_types[_.name]["type"] = _.name
-            campaign_types[_.name]["feature"] = _.feature
-            campaign_types[_.name]["tags"] = {}
-            for x in _.attributes:
-                campaign_types[_.name]['tags'][x.attribute_name] = []
         if request.method == 'GET':
-            form = CampaignForm()
-            form.name.data = campaign_obj.name
-            form.campaign_managers.data = managers
-            form.remote_projects.data = campaign_obj.remote_projects
-            form.types.data = campaign_types
-            form.description.data = campaign_obj.description
-            form.geometry.data = json.dumps(
-                get_campaign_geometry(campaign_obj)
-                )
-            form.map_type.data = campaign_obj.map_type
-            form.selected_functions.data = json.dumps(
-                get_campaign_function(
-                    campaign_obj.functions
-                    )
-                )
-            form.start_date.data = campaign_obj.start_date
-            if campaign_obj.end_date:
-                form.end_date.data = campaign_obj.end_date
+            context, form = get_edit_copy_arguments(campaign_obj)
```

Comment:
I feel we could find another name for the function get_edit_copy_arguments?
what does the function do? retrieve managers, get the campaign types, fill up a context hash and initialize a CampaignForm object. That's too much for a function.